tests/csapi/room_leave_test.go

```diff
@@ -0,0 +1,262 @@
+			"msgtype": "m.text",
+			"body":    beforeMessageOne,
+		},
+	})
+
+	alice.SendEventSynced(t, roomID, b.Event{
+		Type: "m.room.message",
+		Content: map[string]interface{}{
+			"msgtype": "m.text",
+			"body":    beforeMessageTwo,
+		},
+	})
+
+	bob.LeaveRoom(t, roomID)
+
+	alice.MustSyncUntil(t, client.SyncReq{}, client.SyncLeftFrom(bob.UserID, roomID))
+
+	_, bobSinceToken := bob.MustSync(t, client.SyncReq{TimeoutMillis: "0"})
+
+	alice.SendEventSynced(t, roomID, b.Event{
+		Type:     "m.room.name",
+		StateKey: b.Ptr(""),
+		Content: map[string]interface{}{
+			"name": afterRoomName,
+		},
+	})
+
+	alice.SendEventSynced(t, roomID, b.Event{
+		Type:     madeUpStateKey,
+		StateKey: b.Ptr(""),
+		Content: map[string]interface{}{
+			"body": afterMadeUpState,
+		},
+	})
+
+	alice.SendEventSynced(t, roomID, b.Event{
+		Type: "m.room.message",
+		Content: map[string]interface{}{
+			"msgtype": "m.text",
+			"body":    afterMessage,
+		},
+	})
+
+	// Have charlie join the room, to check against /members calls later
+	charlie.JoinRoom(t, roomID, nil)
+	alice.MustSyncUntil(t, client.SyncReq{}, client.SyncJoinedTo(charlie.UserID, roomID))
+
+	// sytest: Can get rooms/{roomId}/state for a departed room (SPEC-216)
+	t.Run("Can get rooms/{roomId}/state for a departed room", func(t *testing.T) {
+		// Bob gets the old state
+		resp := bob.MustDoFunc(
+			t,
+			"GET",
+			[]string{"_matrix", "client", "v3", "rooms", roomID, "state", madeUpStateKey},
+		)
+		must.MatchResponse(t, resp, match.HTTPResponse{
+			JSON: []match.JSON{
+				match.JSONKeyEqual("body", beforeMadeUpState),
+			},
+		})
+
+		// ...While Alice gets the new state
+		resp = alice.MustDoFunc(
+			t,
+			"GET",
+			[]string{"_matrix", "client", "v3", "rooms", roomID, "state", madeUpStateKey},
+		)
+		must.MatchResponse(t, resp, match.HTTPResponse{
+			JSON: []match.JSON{
+				match.JSONKeyEqual("body", afterMadeUpState),
+			},
+		})
+	})
+
+	// sytest: Can get rooms/{roomId}/members for a departed room (SPEC-216)
+	t.Run("Can get rooms/{roomId}/members for a departed room", func(t *testing.T) {
+		resp := bob.MustDoFunc(
+			t,
+			"GET",
+			[]string{"_matrix", "client", "v3", "rooms", roomID, "members"},
+		)
+
+		must.MatchResponse(t, resp, match.HTTPResponse{
+			JSON: []match.JSON{
+				match.JSONCheckOff("chunk",
+					[]interface{}{
+						"m.room.member|" + alice.UserID + "|join",
+						"m.room.member|" + bob.UserID + "|leave",
+					}, func(result gjson.Result) interface{} {
+						return strings.Join([]string{
+							result.Map()["type"].Str,
+							result.Map()["state_key"].Str,
+							result.Get("content.membership").Str,
+						}, "|")
+					}, nil),
+			},
+		})
+	})
+
+	// sytest: Can get rooms/{roomId}/messages for a departed room (SPEC-216)
+	t.Run("Can get rooms/{roomId}/messages for a departed room", func(t *testing.T) {
+		resp := bob.MustDoFunc(t, "GET", []string{"_matrix", "client", "v3", "rooms", roomID, "messages"}, client.WithQueries(url.Values{
```

Comment:
```suggestion
		// TODO: try this with the most recent since token too
		resp := bob.MustDoFunc(t, "GET", []string{"_matrix", "client", "v3", "rooms", roomID, "messages"}, client.WithQueries(url.Values{
```